src/Features/Core/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer.NestedTypes.cs

```diff
@@ -102,12 +102,12 @@ public override bool Equals(object obj)
                     return false;
                 }
 
-                return ProviderId == other.ProviderId && StateTypeId == other.StateTypeId && Key == other.Key;
+                return Analyzer == other.Analyzer && StateTypeId == other.StateTypeId && Key == other.Key;
             }
 
             public override int GetHashCode()
             {
-                return Hash.Combine(Key, Hash.Combine(ProviderId, (int)StateTypeId));
+                return Hash.Combine(Key, Hash.Combine(Analyzer.GetHashCode(), (int)StateTypeId));
```

Comment:
Callback into analyzer? This can throw right? Also what guarantees that 2 distinct analyzer instances will not return same value, its all user code.
Can we probably use fully qualified name of Analyzer + version of containing analyzer reference or something?